### PR TITLE
Feature/myMedication

### PR DESCRIPTION
**Pull request by S1rDan**
Medications Feature.

- Create new medications
- Edit and Remove medications
- Screen Routing
- Centralized Theming

### CODE REVIEW

**Comment by inchw0rm** [comment]
Ok I'm not going to write it in every file, but the whitespaces are inconsistent, some lines use 2 spaces some lines use 4. Please fix this, so far the project uses 2 spaces for indentation and let's keep this consistent. You may want to run `dart format` on your code.

**Comment by S1rDan** [comment]
> Ok I'm not going to write it in every file, but the whitespaces are inconsistent, some lines use 2 spaces some lines use 4. Please fix this, so far the project uses 2 spaces for indentation and let's keep this consistent. You may want to run `dart format` on your code.

`dart format` didnt worked. Maybe it's fixed with the merge.

File: lib/screens/add_medication_screen/add_medication_screen.dart [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+  double _night = 0;
+
+  /// The compound to be taken.
+  String? _compound;
+
+  /// Determines if you are able to add the medication.
+  bool _addDisabled = true;
+
+  Expanded _createAmountSelect(
+      EdgeInsets padding, String hint, String medicationTime) {
```
Comment (QuakeNoob) [inline lib/screens/add_medication_screen/add_medication_screen.dart:43]: Can you try to add a trailing comma behind "String medicationTime"?

File: lib/screens/main_screen/main_screen.dart [hunk 2]
```diff
@@ -28,17 +35,51 @@ class MainScreen extends StatelessWidget {
...
+                ),
+              ),
+            ),
+            home: const BottomBar(title: 'PICOS'),
+            routes: <String, Widget Function(BuildContext)>{
+              '/my-medications': (BuildContext ctx) =>
+                  const MyMedicationsScreen(),
+              '/add-medication': (BuildContext ctx) =>
+                  const AddMedicationScreen()
+            }),
```
Comment (QuakeNoob) [inline lib/screens/main_screen/main_screen.dart:81]: Please split here }), is better to have ")," in a new line and at the same level as the corosponding "(" in line 55

File: lib/screens/questionaire_screen/questionaire_card.dart [hunk 3]
```diff
@@ -23,15 +23,14 @@ class _QuestionaireCardState extends State<QuestionaireCard> {
...
-          setState(() {
-            _selectedElement = newSelection;
+          title: Text(answer.value),
+          value: answer.key,
+          groupValue: _selectedElement,
+          onChanged: (int? newSelection) {
+            setState(() {
+              _selectedElement = newSelection;
+            });
           });
```
Comment (QuakeNoob) [inline lib/screens/questionaire_screen/questionaire_card.dart:33]: Is here a trailing comma between } and ) possible?

File: lib/states/medications_list_bloc.dart [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import 'package:picos/repository/medications_repository.dart';
+
+import '../models/medication.dart';
+
+part 'medications_list_event.dart';
+
+part 'medications_list_state.dart';
+
+/// Bloc for gluing MedicationsListEvents and MedicationsListState together.
+class MedicationsListBloc
```
Comment (QuakeNoob) [inline lib/states/medications_list_bloc.dart:14]: is this name intentional or is this a typo "MedicationsListBlock" ?

File: lib/states/medications_list_bloc.dart [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+    on<MedicationsListSubscriptionRequested>(_onSubscriptionRequested);
+    on<SaveMedication>(_onSaveMedication);
+    on<RemoveMedication>(_onRemoveMedication);
+  }
+
+  final MedicationsRepository _medicationsRepository;
+
+  Future<void> _onSubscriptionRequested(
+      MedicationsListSubscriptionRequested event,
+      Emitter<MedicationsListState> emit) async {
```
Comment (QuakeNoob) [inline lib/states/medications_list_bloc.dart:29]: trailing comma behind "emit" ?

**Comment by S1rDan** [comment]
Should this PR get closed since all changes are merged with @mustiwebp repository?

**Comment by mustiwebp** [comment]
> Should this PR get closed since all changes are merged with @mustiwebp repository?

That's fine with me. I will open another PR.

**Pull request closed without merging**